### DISCUSSION TITLE
Deleting or renaming files in a multi-file Gist using Github API

### DISCUSSION THREAD

**Discussion by vadimkantorov** [community/community#24775] [Apps, API and Webhooks]
<p>The docs at <a href="https://docs.github.com/en/rest/reference/gists#update-a-gist" class="inline-onebox" rel="noopener nofollow ugc">Gists - GitHub Docs</a> mention that PATCH method can be used for <code>update or delete a gist file and rename gist files</code>.</p>
<ul>
<li>
<p>How do I express deleting a file inside a Gist? Should I specify <code>content : null</code>?</p>
</li>
<li>
<p>How do I express renaming a file?</p>
</li>
</ul>

**Comment by UziTech** [comment, marked as answer]
<blockquote>
<p>How do I express deleting a file inside a Gist?</p>
</blockquote>
<p>Delete a file by setting <code>content</code> to an empty string (<code>""</code>).</p>
<pre><code class="lang-json">{
  "files": {
    "deleteme.md": {
      "content": ""
    }
  }
}
</code></pre>
<p>or not setting any properties.</p>
<pre><code class="lang-json">{
  "files": {
    "deleteme.md": {}
  }
}
</code></pre>
<blockquote>
<p>How do I express renaming a file?</p>
</blockquote>
<p>Rename a file by setting the <code>filename</code> property.</p>
<pre><code class="lang-json">{
  "files": {
    "renameme.md": {
      "filename": "renamed.md"
    }
  }
}
</code></pre>
<p>source: <a href="https://github.com/octokit/rest.js/issues/19#issuecomment-824203203" class="inline-onebox" rel="noopener nofollow ugc">Cannot delete gist file with filename: null · Issue #19 · octokit/rest.js · GitHub</a></p>
